### PR TITLE
Remove legacy references (part 1)

### PR DESCRIPTION
**Pull request by matthew-src**
#### Summary

This PR removes references to legacy versions of Mattermost.

### CODE REVIEW

File: source/configure/configuation-in-a-database.rst [hunk 1]
```diff
@@ -1,215 +0,0 @@
-Configuration in a database
```
Comment (cwarnermm) [inline source/configure/configuation-in-a-database.rst:1]: Please revert this file deletion, then replace the introductory sentence to: "You can use the database as the single source of truth..."

File: source/configure/configuration-settings.rst [hunk 2]
```diff
@@ -26,14 +26,6 @@ Mattermost configuration settings are organized into the following categories wi
 
 In self-hosted Mattermost deployments, configuration settings are maintained in the ``config.json`` configuration file, located in the ``mattermost/config`` directory, or `stored in the database <https://docs.mattermost.com/configure/configuation-in-a-database.html>`__. System Admins managing self-hosted workspaces can also modify the ``config.json`` file directly using a text editor.
 
-Configuration in database
```
Comment (cwarnermm) [inline source/configure/configuration-settings.rst:29]: Please revert this deletion, then update the first sentence to: "Self-hosted system configuration can be stored in the database."

File: source/configure/site-configuration-settings.rst [hunk 3]
```diff
@@ -196,9 +194,7 @@ Privacy Policy link
 |                                                                                                                                                                                                    | - ``config.json`` setting: ``.SupportSettings.PrivacyPolicyLink`` |
 | String input. Default is ``https://about.mattermost.com/default-privacy-policy/``.                                                                                                                 | - Environment variable: ``MM_SUPPORTSETTINGS_PRIVACYPOLICYLINK``  |
 +----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------+
-| **Note**: From Mattermost v5.17, this setting does not change the **Privacy Policy** link in the **About Mattermost** window.                                                                                                                                          |
```
Comment (cwarnermm) [inline source/configure/site-configuration-settings.rst:199]: Please revert this line removal and update the note to say that the setting doesn't change the privacy policy link

File: source/configure/web-server-configuration-settings.rst [hunk 4]
```diff
@@ -470,13 +470,11 @@ Trusted proxy IP header
 +--------------------------------------------------------+------------------------------------------------------------------------------+
 | **Notes**:                                                                                                                            |
 |                                                                                                                                       |
-| - From Mattermost v5.12, new deployments set this value to ``[]``, meaning that no header will be trusted. Prior to v5.12, the        |
```
Comment (cwarnermm) [inline source/configure/web-server-configuration-settings.rst:473]: Please revert this line removal and start the sentence with "The default value of ``[]`` means that no header will be trusted."

**Comment by matthew-src** [comment]
Hi @cwarnermm, I believe the latest commits address all of the comments.

File: source/onboard/migrating-to-mattermost.rst [hunk 5]
```diff
@@ -219,35 +219,6 @@ Debugging imports
 
 The ``mmctl import job show`` shows a detailed error message. If you run into problems which the error message does not help to resolve, your best bet is to use the ``mattermost bulk import`` command. The ``mmctl`` import process does not give you any additional debugging information, even in the Mattermost server logs.
 
-Use the Mattermost web app
```
Comment (cwarnermm) [inline source/onboard/migrating-to-mattermost.rst:222]: In this section, we need to keep only some of the information under a subheading of **Migrate from Slack**:
- migrating from Slack requires the `mmetl tool and bulk loading tool </onboard/migrating-to-mattermost.html#migrating-from-slack-using-the-mattermost-mmetl-tool-and-bulk-import>`__.
- the 3 steps: generating a Slack export file, importing from Slack, and selecting the Slack export file

All other details, including when things were deprecated and using the CLI to import can safely be removed.

**Review by cwarnermm — CHANGES_REQUESTED** [review]
Thanks, @matthew-src. I've identified one more round of changes needed on this PR.

File: source/onboard/migrating-to-mattermost.rst (hunk 5 above)
Comment (matthew-src) [reply source/onboard/migrating-to-mattermost.rst]: I think the latest commit addresses this by reverting the deletion of the web app section. Is that alright?

Comment (cwarnermm) [reply source/onboard/migrating-to-mattermost.rst]: Yes. You're right -- your latest commit removes the depreciated CLI content.

**Review by cwarnermm — APPROVED** [review]

**Pull request merged**